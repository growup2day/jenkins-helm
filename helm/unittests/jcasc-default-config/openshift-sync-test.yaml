templates:
  - config-jcasc.yaml
tests:
  - it: should include OpenShift sync in default config script when configured
    set:
      JCasC:
        defaultConfig: true
        openshiftSync:
          enabled: true
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-openshift-sync.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-openshift-sync.yaml]
  - it: should disable OpenShift sync in default config script when not configured
    set:
      JCasC:
        defaultConfig: true
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-openshift-sync.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-openshift-sync.yaml]
  - it: should not include OpenShift sync in default config scripts when there is custom config script for the section
    set:
      JCasC:
        defaultConfig: true
        configScripts:
          custom-override: |
            unclassified:
              globalPluginConfiguration:
                enabled: false
    asserts:
      - isNull:
          path: data.[jcasc-default-config-openshift-sync.yaml]
      - isNotNull:
          path: data.[custom-override.yaml]
