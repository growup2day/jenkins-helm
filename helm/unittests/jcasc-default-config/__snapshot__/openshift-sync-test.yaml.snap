should disable OpenShift sync in default config script when not configured:
  1: |
    |-
      unclassified:
        # Configure System > OpenShift Jenkins Sync
        globalPluginConfiguration:
          enabled: false
should include OpenShift sync in default config script when configured:
  1: |
    |-
      unclassified:
        # Configure System > OpenShift Jenkins Sync
        globalPluginConfiguration:
          enabled: true
